core/libcamera_encoder.hpp

```diff
@@ -33,6 +33,7 @@ class LibcameraEncoder : public LibcameraApp
 	// This is callback when the encoder gives you the encoded output data.
 	void SetEncodeOutputReadyCallback(EncodeOutputReadyCallback callback) { encode_output_ready_callback_ = callback; }
 	void SetMetadataReadyCallback(MetadataReadyCallback callback) { metadata_ready_callback_ = callback; }
+	Encoder *GetEncoder() const { return encoder_.get(); }
```

Comment:
Should this be both public and const?

Perhaps the answer is yes; but if encoder_ is an "owned" object, then it's debatable whether it should be modifiable.

I suppose the gold-plated version would declare const and non-const methods, and have the const one return a `const Encoder*`?

Reply:
In context I think this seems OK to me. Having const functions returning pointers to things that are not const is normal enough (for better or worse!).